src/missingarray.jl

```diff
@@ -26,3 +26,13 @@ Base.fill!(A::CategoricalArray{>:Missing}, ::Missing) = (fill!(A.refs, 0); A)
 
 in(x::Missing, y::CategoricalArray) = false
 in(x::Missing, y::CategoricalArray{>:Missing}) = !all(v -> v > 0, y.refs)
+
+function Missings.replace(a::CategoricalArray{S, N, R, V, C}, replacement::V) where {S, N, R, V, C}
+    pool = CategoricalPool(a.pool)
+    v = C(get!(pool, replacement), pool)
+    Missings.replace(a, v)
+end
+
+function collect(r::Missings.EachReplaceMissing{CategoricalArray{S, N, R, C, V, U},V}) where {S, N, R, C, V, U}
+           CategoricalArray(C[v for v in r])
```

Comment:
Incorrect indentation.

I'm afraid this isn't going to be very efficient. You should be able to get much better performance with something like this (untested, may need to specify type parameters):
```julia
CategoricalArray([v.ref for v in r], r.itr.pool)
```

Reply:
Hmm, `r.itr.pool` won't refer to the right pool, so you'll probably need to use `r.replacement.pool` instead.

Reply:
make sense, if we can assume that the new pool is only longer so that levels in the old pool are still valid in the new one.